app/src/main/java/org/mozilla/fenix/FeatureFlags.kt

```diff
@@ -12,6 +12,12 @@ import mozilla.components.support.locale.LocaleManager.getSystemDefault
  * A single source for setting feature flags that are mostly based on build type.
  */
 object FeatureFlags {
+
+    /**
+     * Enables custom extension collection feature
+     * */
+    val customExtensionCollectionFeature = Config.channel.isNightlyOrDebug || Config.channel.isBeta
+
```

Comment:
The logic to enable the feature is extracted to a new feature flag for better readability and maintenance for the feature. Worth to note that the feature does not only depend on the build variant, there are other conditions that need to be met in order to make the feature visible to the user.

Reply:
> The logic to enable the feature is extracted to a new feature flag for better readability and maintenance for the feature.

💯 

> Worth to note that the feature does not only depend on the build variant, there are other conditions that need to be met in order to make the feature visible to the user.

Do you mean the `amoCollectionOverrideConfigured`/`showSecretDebugMenuThisSession`? This is a good note to add in the kdoc above as well.

Maybe something like, "This feature also requires the AMO Collection override to be enabled (which itself it behind the Secret Settings)."

This is just a nit, so it's up to you. 🙂